nmclicktests/environment.js

```diff
@@ -0,0 +1,269 @@
+        if (!this.hasStarted) {
+          return Bluebird.resolve()
+            .delay(50)
+            .then(() => { return this.ensureStarted(); });
+        }
+      });
+  }
+
+  init(isSkipTempCreate) {
+    return this.getPort()
+      .then(() => this.startServer())
+      .then(() => this.ensureStarted())
+      .timeout(7000)
+      .catch((err) => { throw new Error("Cannot confirm ungit start!!"); })
+  }
+
+  createRepos(config) {
+    return Bluebird.map(config, (conf) => {
+      conf.bare = !!conf.bare;
+      return this.nm.ug.initRepo(conf)
+        .then(() => this.createCommits(conf, conf.initCommits))
+        .then(() => conf.path);
+    });
+  }
+
+  shutdown(doNotClose) {
+    this.shuttinDown = true;
+    return this.nm.ug.backgroundAction('POST', `${rootUrl}/api/testing/cleanup`, undefined)
+      .ug.shutdownServer()
+      .then(() => {
+        if (!doNotClose) {
+          this.nm.end();
+        }
+      });
+  }
+
+  createCommits(config, limit, x) {
+    x = x || 0
+    if (!limit || limit < 0 || x === limit) return Bluebird.resolve();
+
+    return this.nm.ug.createTestFile(`${config.path}/testy${x}`)
+      .then(() => {
+        return this.nm.ug.backgroundAction('POST', `${rootUrl}/api/commit`, {
+          path: config.path,
+          message: `Init Commit ${x}`,
+          files: [{ name: `testy${x}` }]
+        });
+      }).then(() => this.createCommits(config, limit, x + 1))
+  }
+
+  goto(url) {
+    this.nm = this.nm.goto(url);
+    return this.nm;
+  }
+
+  startServer() {
+    console.log('Starting ungit server...', this.config.serverStartupOptions);
+
+    this.hasStarted = false;
+    const options = ['bin/ungit',
+      '--cliconfigonly',
+      `--port=${this.port}`,
+      `--rootPath=${this.config.rootPath}`,
+      '--no-launchBrowser',
+      '--dev',
+      '--no-bugtracking',
+      '--no-sendUsageStatistics',
+      `--autoShutdownTimeout=${this.config.serverTimeout}`,
+      '--logLevel=debug',
+      '--maxNAutoRestartOnCrash=0',
+      '--no-autoCheckoutOnBranchCreate',
+      '--alwaysLoadActiveBranch',
+      '--logGitCommands']
+      .concat(this.config.serverStartupOptions);
+    const ungitServer = child_process.spawn('node', options);
+    ungitServer.stdout.on('data', (data) => {
+      if (this.config.showServerOutput) console.log(prependLines('[server] ', data.toString()));
+
+      if (data.toString().indexOf('Ungit server already running') >= 0) {
+        console.log('server-already-running');
+      }
+
+      if (data.toString().indexOf('## Ungit started ##') >= 0) {
+        if (this.hasStarted) {
+          console.log('Ungit started twice, probably crashed.');
+        } else {
+          this.hasStarted = true;
+          console.log('Ungit server started.');
+        }
+      }
+    });
+    ungitServer.stderr.on("data", (data) => {
+      console.log(prependLines('[server ERROR] ', data.toString()));
+      if (data.indexOf("EADDRINUSE") > -1) {
```

Comment:
In the method above you are always calling `toString` before calling `indexOf` maybe you should do it here too? (And use a separate variable.)